newIDE/app/src/ResourcesList/LocalResourceSources.js

```diff
@@ -220,6 +283,19 @@ const selectLocalResourcePath = (
       },
       paths => {
         if (!paths) return resolve([]);
+        // Update stored preferences values
+        if (paths[0] !== projectPath) {
+          if (values) {
+            if (values.lastOpenedPath[projectName])
+              values.lastOpenedPath[projectName][kind] = paths[0];
+            else {
+              let path = {};
+              path[kind] = paths[0];
+              values.lastOpenedPath[projectName] = path;
+            }
+            savePreferencesValues(values);
+          }
```

Comment:
All of this should not be the responsibility of LocalResourceSources :)
It should be a single function, something like `saveLatestPath(kind, paths[0])`

LocalResourceSources job is to show you dialogs to open files. We want to add it the ability to remember the latest opened path.
For this, we'll pass to it a function to read the last opened path, and another to save it - nothing more. These values and function are the responsibility of the parent to set... and we'll do it using a React context :)

Reply:
Just one confusion is their a way in `PreferencesProvider` to get the project name or I need to pass it with function `saveLatestPath` ?

Reply:
I am thinking that the project name should be passed as an argument to `saveLatestPath` because if we have multiple projects open in future, then Provider needs to know, which project we are talking about. :)

Reply:
That seems fair to me to pass the project name :) Or maybe the project file path?